packages/jest-expect/src/types.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type {AsymmetricMatchers, BaseExpect, Matchers} from 'expect';
+import type {SnapshotState, addSerializer} from 'jest-snapshot';
+
+export type JestExpect = {
+  <T = unknown>(actual: T): JestMatchers<void, T> &
+    Inverse<JestMatchers<void, T>> &
+    PromiseMatchers<T>;
+  addSnapshotSerializer: typeof addSerializer;
+} & BaseExpect &
+  AsymmetricMatchers &
+  Inverse<Omit<AsymmetricMatchers, 'any' | 'anything'>>;
+
+type Inverse<Matchers> = {
+  /**
+   * Inverse next matcher. If you know how to test something, `.not` lets you test its opposite.
+   */
+  not: Matchers;
+};
+
+type JestMatchers<R extends void | Promise<void>, T> = Matchers<R> &
+  SnapshotMatchers<R, T>;
+
+type PromiseMatchers<T = unknown> = {
+  /**
+   * Unwraps the reason of a rejected promise so any other matcher can be chained.
+   * If the promise is fulfilled the assertion fails.
+   */
+  rejects: JestMatchers<Promise<void>, T> &
+    Inverse<JestMatchers<Promise<void>, T>>;
+  /**
+   * Unwraps the value of a fulfilled promise so any other matcher can be chained.
+   * If the promise is rejected the assertion fails.
+   */
+  resolves: JestMatchers<Promise<void>, T> &
+    Inverse<JestMatchers<Promise<void>, T>>;
+};
+
+interface SnapshotMatchers<R extends void | Promise<void>, T> {
```

Comment:
Should live in `jest-snapshot`?

Reply:
yeah, I think so! And then `@jest/expect` adds it to its `JestExpect` type in some way